doc/source/ray-observability/user-guides/profiling.md

```diff
@@ -0,0 +1,56 @@
+(profiling)=
+# Profiling
+Profiling is one of the most important debugging tools to diagnose performance, out of memory, hanging, or other application issues.
+Here is a list of common profiling tools you may use when debugging Ray applications. 
+- CPU profiling
+    - py-spy
+- Memory profiling
+    - memray
+- GPU profiling
+    - PyTorch Profiler
+- Ray Task / Actor profiling
+    - Ray Timeline
+
+If Ray doesn't work with certain profiling tools, try running them without Ray to debug the issues.
+
+## CPU profiling
+CPU profiling for Driver and Worker processes. This helps you understand the CPU usage by different processes and debug unexpectedly high or low usage.
+
+### py-spy
+[py-spy](https://github.com/benfred/py-spy/tree/master) is a sampling profiler for Python programs. It lets you visualize what your Python program is spending time on without restarting the program or modifying the code in any way. 
```

Comment:
There are 2 things it does./

1. It lets you visualize what your Python program is spending time on without restarting the program or modifying the code in any way
2. It dumps the stacktrace of the running process so that you can see what the process is doing at a certain time. It is useful when programs hang

Reply:
Added